### PR TITLE
My rewrite that makes this thing actually work! How about that?

### PR DESCRIPTION
**Pull request by Monadic-Cat**
Your repo failed to build. I made it work. :)

### CODE REVIEW

**Comment by Monadic-Cat** [comment]
I should note that this PR conflicts with #4.

File: Makefile [hunk 1]
```diff
@@ -1,38 +1,25 @@
...
+		cd ../.."
+build/hello_cpp.a: build/hello_cpp.o
+	$(AR) rcs build/hello_cpp.a build/hello_cpp.o
+build/hello_cpp.o: src/hello.cpp
+	$(CXX) -c src/hello.cpp -o build/hello_cpp.o
+
+.PHONY:
+clean: clean_rust
+	rm -rf build
 	mkdir -p build
```
Comment (cbpudding) [inline Makefile]: Couldn't you simply remove the contents with a wildcard?

File: Makefile [hunk 2]
```diff
@@ -1,38 +1,25 @@
...
-
-.PHONY: build
-build: build/ $(OBJECTS)
-
-.PHONY: clean
-clean:
-	$(RM) $(OBJECTS) hello
-
-build/:
+.PHONY:
```
Comment (cbpudding) [inline Makefile:1]: I didn't know you could use .PHONY like that

File: Makefile [hunk 3]
```diff
@@ -1,38 +1,25 @@
...
+run: build/hello
+	./build/hello
+
+build/hello: build/librust_hello.a build/hello_cpp.a
+	$(CC) src/main.c src/hello.c build/hello_cpp.a build/librust_hello.a -o build/hello -lstdc++
+build/librust_hello.a: src/rust-hello/*
+	sh -c "cd src/rust-hello &&\
+		cargo build --release &&\
+		cp target/release/librust_hello.a ../../build/librust_hello.a &&\
+		cd ../.."
```
Comment (cbpudding) [inline Makefile]: I don't like using sh in a Makefile

File: src/main.c [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#include <functions.h>
+#include "../include/functions.h"
```
Comment (cbpudding) [inline src/main.c:1]: We have -I for a reason

File: src/rust-hello/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#![no_std]
+
+use core::panic::PanicInfo;
+use libc::{abort, puts};
```
Comment (cbpudding) [inline src/rust-hello/src/lib.rs:4]: What's the point of using Rust if you have to use libc?

**Review by cbpudding — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (Monadic-Cat) [reply Makefile]: I could. But then, this is more complete. There might be directories in `build` at some point, so I'll need to `-r` anyway.

File: Makefile (hunk 2 above)
Comment (Monadic-Cat) [reply Makefile:1]: It might be specific to GNU Make, I'm not sure. Let me know if there's another Make you want to support.

File: Makefile (hunk 3 above)
Comment (Monadic-Cat) [reply Makefile]: I don't either, but Cargo requires you be under the same directory as the crate you're working with.

File: src/main.c (hunk 4 above)
Comment (Monadic-Cat) [reply src/main.c:1]: That's true. I was just in a hurry to make something work. I'll probably fix this tonight.

File: src/rust-hello/src/lib.rs (hunk 5 above)
Comment (Monadic-Cat) [reply src/rust-hello/src/lib.rs:4]: It's what we have to do in order to statically link Stable Rust into C. When that changes, so shall this.

It's not all bad, though. There are still crates we can pull in to make things nicer.

Comment (cbpudding) [reply src/rust-hello/src/lib.rs:4]: If we can't use Rust's standard library, then what's the point of using Rust?

Comment (Monadic-Cat) [reply src/rust-hello/src/lib.rs:4]: Macros, better types, memory safety, `async`/`await`, it looks nicer, ...

**Comment by Monadic-Cat** [comment]
Closing in favor of #4.

**Pull request closed without merging**